### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/presentable-error#2]
General discussion about how this works.

Continued from https://github.com/sindresorhus/del-cli/issues/34.

**Comment by tommy-mitchell** [comment]
It might also be useful to mark the name of the module throwing the error, in case a consumer only wants to present errors from a specific module:

```js
try {
  del(path);
} catch (error) {
  if (isPresentableError(error) && error.module === 'del') {
    console.error(error.message);
    process.exit(1);
  }
  
  throw error;
}

**Comment by sindresorhus** [comment]
@tommy-mitchell Yes, maybe. But I think we need some more real-world experience with this before adding additional properties. For example, app code may want this too, not just modules. So a free-form `.presentableSender` property may be more flexible.

**Comment by emanuel-lundman** [comment]
In one of our projects we’ve got both a public message (presentable) and an optional more tech message. That we can throw. In that way we can present the presentable part to end users but still have more details for the errors that should both be presentable and handled by the dev team. The presentable message could in our case be multilingual when we want to as well. 

It’s a bit of different approach than have an error only thrown to be presentable.
Just wanted to add it to the conversation as feedback.

TLDR; We’ve seen the use of an error that can be both presentable and include more details for the developers.

**Comment by tom-sherman** [comment]
> We’ve seen the use of an error that can be both presentable and include more details for the developers.

Came here to suggest this. Instead of a Boolean property being the convention, how about a method like `presentError` that returns a string.

In the simple case this could just `return this.message` (this library could automate this plumbing) and in the more complex case you can implement it yourself.
